contracts/BoostedMasterChefJoe.sol

```diff
@@ -0,0 +1,621 @@
+        if (address(_rewarder) != address(0)) {
+            _rewarder.onJoeReward(msg.sender, user.amount);
+        }
+
+        pool.lpToken.safeTransfer(msg.sender, amount);
+
+        emit Withdraw(msg.sender, pid, amount);
+    }
+
+    /// @notice Updates a pool and user boost factors after
+    /// a veJoe token operation. This function needs to be called
+    /// by the veJoe contract after every event.
+    /// @param _user The users address we are updating
+    /// @param _newUserBalance The new balance of the users veJoe
+    function updateBoost(address _user, uint256 _newUserBalance) external {
+        require(msg.sender == address(VEJOE), "Caller not veJOE");
+
+        uint256 length = poolInfo.length;
+
+        for (uint256 pid = 0; pid < length; ++pid) {
+            UserInfo storage user = userInfo[pid][_user];
+
+            // Skip if user doesn't have any deposit in the pool
+            if (user.amount == 0) {
+                continue;
+            }
+
+            PoolInfo storage pool = poolInfo[pid];
+
+            updatePool(pid);
+
+            // calculate pending
+            uint256 pending = user
+                .amount
+                .mul(getUserBoost(pid, _user))
+                .mul(pool.accJoePerShare)
+                .div(ACC_TOKEN_PRECISION)
+                .sub(user.rewardDebt);
+
+            // Increase claimableJoe
+            claimableJoe[pid][_user] += pending;
+
+            // Update pool.totalVeJoe
+            pool.totalVeJoe = pool.totalVeJoe.sub(user.veJoeBalance).add(_newUserBalance);
+
+            // Decrement the total boosted amount and increment it after updating the
+            // users veJoeBalance .
+            pool.totalBoostedAmount = pool.totalBoostedAmount.sub(user.amount.mul(getUserBoost(pid, _user)));
+
+            // Update users veJoeBalance.
+            user.veJoeBalance = _newUserBalance;
+
+            // Update the pools total boosted amount.
+            pool.totalBoostedAmount = pool.totalBoostedAmount.add(user.amount.mul(getUserBoost(pid, _user)));
+            // update reward debt, take into account new boost
+            user.rewardDebt = user
+                .amount
+                .mul(getUserBoost(pid, _user))
+                .mul(pool.accJoePerShare)
+                .div(ACC_TOKEN_PRECISION)
+                .div(1000);
+        }
+    }
+
+    /// @notice Returns a users boosted emission share
+    /// @param _pid The index of the pool. See `poolInfo`
+    /// @param _user The address of the depositor
+    /// @dev `userBoost` is scaled by 1e3
+    function getUserBoost(uint256 _pid, address _user) public view returns (uint256 userBoost) {
+        PoolInfo memory pool = poolInfo[_pid];
+        UserInfo memory user = userInfo[_pid][_user];
+        userBoost = 1000;
+        if (user.veJoeBalance > 0 && pool.totalVeJoe > 0) {
+            userBoost = userBoost.add(user.veJoeBalance.mul(maxBoostFactor).div(pool.totalVeJoe));
```

Comment:
Why not divide by 1000 here instead?

Reply:
The reason for that is:
example:
```
userBoost = 1000
user.veJoeBalance = 10
pool.totalVeJoe = 10
maxBoostFactor = 1500
1000 + (10 * 1500 / 10)
1000 + (1500)
2500
```
If I divide this by 1000 here the value will round to `2` where really it should be `2.5`.

Reply:
Ah yes good point